From 846ba31664193b4c0fcfd7f6a81d1829ad115269 Mon Sep 17 00:00:00 2001
From: nate zhou <gnuunixchad@outlook.com>
Date: Thu, 26 Mar 2026 16:25:13 +0800
Subject: [PATCH] always use select_bg to draw boxes on tags

---
 src/kwm/bar.zig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/kwm/bar.zig b/src/kwm/bar.zig
index 57ab7ea..17b3674 100644
--- a/src/kwm/bar.zig
+++ b/src/kwm/bar.zig
@@ -500,7 +500,7 @@ fn render_static_component(self: *Self) void {
                 buffer,
                 false,
                 .top,
-                if (is_focused) &transparent else &select_bg,
+                &select_fg,
                 x,
                 y,
             );
-- 
2.53.0

