From e94983fc31839ab56129d6f285517c53319be359 Mon Sep 17 00:00:00 2001
From: nate zhou <gnuunixchad@outlook.com>
Date: Sat, 21 Mar 2026 12:53:20 +0800
Subject: [PATCH] bar: reduce tag labels padding to 3/4

---
 src/kwm/bar.zig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/kwm/bar.zig b/src/kwm/bar.zig
index 30b197e..d8cfbec 100644
--- a/src/kwm/bar.zig
+++ b/src/kwm/bar.zig
@@ -233,7 +233,7 @@ inline fn static_component_width(self: *Self) i32 {
 
 
 inline fn get_pad(self: *Self) u16 {
-    return @intCast(self.font.height);
+    return @intCast(@divFloor(self.font.height * 3, 4)); // reduce padding to 3/4
 }
 
 
-- 
2.53.0

