commit 51383822ab90acce60c54f66f144293dbe7735c1
Author: nate zhou <gnuunixchad@outlook.com>
Date:   Tue Jan 20 17:48:38 2026 +0800

    Mod(bar.zig): reduce tags padding to 0.75

diff --git a/src/kwm/bar.zig b/src/kwm/bar.zig
index bb0d112..fdbd115 100644
--- a/src/kwm/bar.zig
+++ b/src/kwm/bar.zig
@@ -184,7 +184,7 @@ pub fn render(self: *Self) void {
 
 
 inline fn get_pad(self: *Self) u16 {
-    return @intCast(self.height());
+    return @intCast(@divFloor(self.height() * 3, 4)); // 0.75 = 3/4
 }
 
 
